### ISSUE TITLE
Search breaks with '%' in query 

### ISSUE THREAD

**Issue by jvwong** [PathwayCommons/pathways-search#89]
Breaking case

INTEGRIN SIGNALLING PATHWAY%PANTHER PATHWAY%P00034

```javascript
Uncaught URIError: URI malformed
    at decodeURIComponent (<anonymous>)
    at t.value (http://beta.pathwaycommons.org/pathways/bundle.js:67:68383)
    at t.value (http://beta.pathwaycommons.org/pathways/bundle.js:67:68776)
    at onKeyPress (http://beta.pathwaycommons.org/pathways/bundle.js:67:71111)
    at Object.r (http://beta.pathwaycommons.org/pathways/bundle.js:33:21121)
    at a (http://beta.pathwaycommons.org/pathways/bundle.js:33:17209)
    at Object.s [as executeDispatchesInOrder] (http://beta.pathwaycommons.org/pathways/bundle.js:33:17424)
    at d (http://beta.pathwaycommons.org/pathways/bundle.js:16:34672)
    at g (http://beta.pathwaycommons.org/pathways/bundle.js:16:34798)
```

**Comment by d2fong** [comment]
as in searching 'INTEGRIN SIGNALLING PATHWAY%PANTHER PATHWAY%P00034' in the search box?

**Comment by d2fong** [comment]
here is where is breaks:
https://github.com/PathwayCommons/pathways-search/blob/integration/js/search/components/SearchHeader.jsx#L35


Seems like it possibly uses decodeURIComponent to allow people to search using urls?  I am not sure.

To differentiate between a normal text-only query and a url query, it checks if there is a percent in the query:
```
var q = this.state.q.indexOf("%") === -1 
```

**Comment by d2fong** [comment]
I assumed based on the code that it is to handle the user putting in something like:

http://identifiers.org/reactome/R-HSA-111463

But the search app says page not found for queries like that, so I am not sure why that is there.

**Comment by jvwong** [comment]
OK I think I know what happened [here](https://github.com/PathwayCommons/pathways-search/blob/8995f05cee57c5baeaa1d08b2928c22df6baba72/js/search/components/SearchHeader.jsx#L35-L37).

```js
var q = this.state.q.indexOf("%") === -1 ? this.state.q : decodeURIComponent(this.state.q);
		if (q.match(/^(?:http(s)?:\/\/)?[\w.-]+(?:\.[\w\.-]+)+[\w\-\._~:/?#[\]@!\$&'\(\)\*\+,;=.]+$/)) {
			this.props.history.push({pathname: "/view", search: queryString.stringify({uri: q})});
		}
```

Indeed the app should be capable of accepting valid URIs for pathways. 

1.   I didn't switch over to the ```pathname``` to  ```/view`` when I [renamed this component](6e6b124628c06f395c0c7a88023e36d12071b938). This resulted in the page not found. Fixing this now allows valid URIs. 

2. The ```var q = this.state.q.indexOf("%") === -1 ``` is trying to accommodate encoded URIs. This should just accept unencoded so that the second regex match is sufficient.

**Comment by jvwong** [comment]
Fixed in 7074d11020bddca57205596ffd2bc7ee57215923
